overlord/devicestate/devicemgr.go

```diff
@@ -936,6 +936,7 @@ func (m *DeviceManager) ensureSeeded() error {
 	}
 
 	if m.changeInFlight("seed") {
+		m.ensureAutoImportAssertions()
```

Comment:
this is a strange place to do this it means we are during seeding but we might also never get here depending on details

Reply:
I knew this will be challenged :)
There is logic I put it here.
In general, we have challenge to how to debug failed seeding, so creating a user while seeding, opens the option to debug this. The drawback is that this might be checked rather often during seeding...
Thinking about my rationale again, and the fact we might be compressing the whole install mode into the initramfs, my argument for debugging is gone.

So I'd propose moving it once seeding is done, so in `ensureSeeded()`
```
	if seeded {
		m.ensureAutoImportAssertions()
		return nil
	}
```

If to move it to `Ensure()` function with all other "ensure" calls, I'd need to check if seeding is completed, this means repeating
```
	var seeded bool
	err := m.state.Get("seeded", &seeded)
	if err != nil && !errors.Is(err, state.ErrNoState) {
		return err
	}
```
this is repeated in 8 other "ensure***" functions, this almost calls for having `m.seeded` as we have `m.preseed` which be updated in `ensureSeeded()` as it's always called first from `Ensure()`